### DISCUSSION TITLE
R column of the status display

### DISCUSSION THREAD

**Discussion by v74863** [ZerBea/hcxdumptool#432] [General]
Hi, Mike

Could you please explain what makes the plus sign (+) appear in the R column of the status display.
In other words, what events must occur for this.

In the help we read:

` R = + AP display     : AP is in TX range or under attack
`

Looks like the conditions below are responsible for the R column.
```
	if(((aplist +i)->status & AP_IN_RANGE) == AP_IN_RANGE) ar = pmok;
	else ar = pmdef;

```
But I just started learning C language, so please help.
Thanks in advance.

**Comment by ZerBea** [comment]
The "+" appear if the AP respond to hcxdumptool's AUTHENTICATIONREQUEST.
https://github.com/ZerBea/hcxdumptool/blob/master/hcxdumptool.c#L2298
It disappear if hcxdumptool doesn't transmit requests any longer (PMKID and/or EAPOL M1M2M3 have been received).
It disappear if the AP dos not respond to hcxdumptool's requests (AUTHENTICATIONREQUEST).

That also depends on the elapsed time since we got the last response:
https://github.com/ZerBea/hcxdumptool/blob/master/hcxdumptool.c#L605

This column will be removed in one of the next versions, because we go full stealth.
As soon as a CLIENT has been received, hcxdumptool will use the MAC of the CLIENT instead its own. Straight after going stealth it is not possible any longer to distinguish between a RESPONSE to hcxdumptool or a response to the CLIENT.
You can test this (new feature) if you compile hcxlabtool.

**Reply by ghost** [reply]
R and S columns not very help, support delet them

**Reply by ZerBea** [reply]
"S" is important, because you should know why this NETWORK is ignored.
 "R" can be removed, because we get the same information from "123P". If "S" is not empty and 123P are empty, AP is not in RANGE or does not respond.

**Comment by ZerBea** [comment]
BTW:
My way of programming is not the standard.
Without knowledge of 802.11 protocol, several functions are very difficult to understand.

**Comment by v74863** [comment]
Thank you for kind reply. Your tools are one of my favorite hobbies. Some people like crosswords, some chess, some prefer huge amounts of beer, and I like digging into these things. Brain training and prevention of upcoming dementia :)

I wonder if it makes sense to use assembly language for the most critical part of hcxdumptool to improve performance? Or the compiled C code is fast enough? I am referring to the purely technical aspect and do not take into account the high labor-intensiveness of writing assembly code.

**Reply by ZerBea** [reply]
Major problem is the firmware, the driver and the raw packet socket.
In AP mode, the device/driver/kernel/application needs to be able to handle one NETWORK.
In MONITOR mode, the device/driver/kernel/application needs to handle the entire traffic.
The code, translated by gcc is not perfect, but better than I ever can do in ASM.
